x-pack/plugins/encrypted_saved_objects/server/saved_objects/saved_objects_encryption_extension.test.ts

```diff
@@ -0,0 +1,204 @@
+      attrThree: 'three',
+    },
+    score: 1,
+    references: [],
+  };
+
+  test('does not alter response if type is not registered', async () => {
+    const { extension, service } = setup();
+
+    expect(extension.decryptOrStripResponseAttributes(unregisteredSO)).resolves.toEqual({
+      ...unregisteredSO,
+      attributes: {
+        attrOne: 'one',
+        attrSecret: 'secret',
+        attrNotSoSecret: 'not-so-secret',
+        attrThree: 'three',
+      },
+    });
+
+    expect(service.decryptAttributes).not.toHaveBeenCalled();
+  });
+
+  test('strips encrypted attributes except for ones with `dangerouslyExposeValue` set to `true` if type is registered', async () => {
+    const { extension, service } = setup();
+
+    expect(extension.decryptOrStripResponseAttributes(registeredSO)).resolves.toEqual({
+      ...registeredSO,
+      attributes: { attrOne: 'one', attrNotSoSecret: 'not-so-secret', attrThree: 'three' },
+    });
+
+    expect(service.stripOrDecryptAttributes).toHaveBeenCalledTimes(1);
+  });
+
+  test('includes both attributes and error if decryption fails.', async () => {
+    const { extension, service } = setup();
+
+    const decryptionError = new EncryptionError(
+      'something failed',
+      'attrNotSoSecret',
+      EncryptionErrorOperation.Decryption
+    );
+    service.stripOrDecryptAttributes.mockResolvedValue({
+      attributes: { attrOne: 'one', attrThree: 'three' },
+      error: decryptionError,
+    });
+
+    expect(extension.decryptOrStripResponseAttributes(unregisteredSO)).resolves.toEqual({
+      ...unregisteredSO,
+      attributes: {
+        attrOne: 'one',
+        attrSecret: 'secret',
+        attrNotSoSecret: 'not-so-secret',
+        attrThree: 'three',
+      },
+    });
+
+    expect(service.stripOrDecryptAttributes).not.toHaveBeenCalled();
+
+    expect(extension.decryptOrStripResponseAttributes(registeredSO)).resolves.toEqual({
+      ...registeredSO,
+      attributes: { attrOne: 'one', attrThree: 'three' },
+      error: decryptionError,
+    });
+
+    expect(service.stripOrDecryptAttributes).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('#encryptAttributes', () => {
+  test('does not encrpyt attributes if type is not registered', async () => {
+    const { extension, service } = setup();
+
+    expect(
+      extension.encryptAttributes(
+        {
+          type: 'unknown-type',
+          id: 'mock-saved-object-id',
+          namespace: undefined,
+        },
+        {
+          attrOne: 'one',
+          attrSecret: 'secret',
+          attrNotSoSecret: 'not-so-secret',
+          attrThree: 'three',
+        }
+      )
+    ).resolves.toEqual({
+      attrOne: 'one',
+      attrSecret: 'secret',
+      attrNotSoSecret: 'not-so-secret',
+      attrThree: 'three',
+    });
+
+    expect(service.encryptAttributes).toBeCalledTimes(1); // ToDo: should the extension even bother calling encrypt if the type is not encryptable? See Decrypt as example.
```

Comment:
It looks like we call encrypt regardless of whether a type is encryptable. Is there a good reason for this?

Reply:
I added the check to the extension function so it will not call the service function if the type is not encryptable.